sphinx/domains/std.py

```diff
@@ -202,7 +204,16 @@ def handle_signature(self, sig: str, signode: desc_signature) -> str:
             if count:
                 signode += addnodes.desc_addname(', ', ', ')
             signode += addnodes.desc_name(optname, optname)
-            signode += addnodes.desc_addname(args, args)
+            if self.env.config.option_parse_variable_part:
+                for part in samp_role.parse(args):
+                    if isinstance(part, nodes.Text):
+                        # in order to preserve whitespaces, call desc_addname
+                        text = part.astext()
+                        signode += addnodes.desc_addname(text, text)
```

Comment:
I guess here one could do something like
```py
if len(text) != 0 and len(text.strip()) == 0:
   signode += addnodes.desc_sig_space()
else:
  signode += addnodes.desc_addname(text, text)
```
but it is more a theoretical exercise at this point I guess.